### ISSUE TITLE
parse dbc with istream instead of filename

### ISSUE THREAD

**Issue by TheWhiteBug** [LinuxDevon/dbc_parser_cpp#27]
Hi Devon! Thanks for the great lib! I selected it because of its simplicity. 

```code
void DbcParser::parse_file(const std::string& file) {
	std::ifstream s(file.c_str());
(...)
```

The `file` argument is only needed for the `ifstream` .
I would suggest to expose an `std::istream` overload in which case hitting the filesystem would not be required.
This looks like a low hanging fruit while the current limitation causes my current project to depend on writable storage which is highly inconvenient.

Thanks in advance!
Also, reading the announcement, you can make it a c++17 project if you ask me :)

**Comment by LinuxDevon** [comment]
Hi @TheWhiteBug, Thank you for the feedback and appreciate you picking my library! That seems like an easy add and can take care of that. I will try and add that in a few weeks. Bit slammed on time right now. I can cut a release with that change when I get that in.

**Comment by LinuxDevon** [comment]
Will be included in `v0.5.0`
